engine/src/main/java/org/apache/apex/stram/DeployManager.java

```diff
@@ -0,0 +1,293 @@
+  {
+    return deployRequests.get(containerId);
+  }
+
+  /**
+   * Returns deploy/undeploy group Id for container
+   * @param containerId
+   * @return groupId <br/>
+   *         <b>Note:</b> groupId 0 indicates and indipendent event, with no
+   *         group
+   */
+  public long getDeployGroupIdForContainer(String containerId)
+  {
+    long groupId = 0;
+    if (deployRequests.get(containerId) != null) {
+      groupId = deployRequests.get(containerId).groupId;
+    }
+    return groupId;
+  }
+
+  /**
+   * Returns deploy/undeploy group Id for container This could be a new
+   * container allocated during redeploy process
+   * @param containerId
+   * @return groupId <br/>
+   *         <b>Note:</b> groupId 0 indicates and indipendent event, with no
+   *         group
+   */
+  public long getDeployGroupIdForAffectedContainer(String containerId)
+  {
+    long groupId = getDeployGroupIdForContainer(containerId);
+    if (groupId != 0) {
+      return groupId;
+    }
+    for (DeployRequest request : getDeployRequests().values()) {
+      if (request.affectedContainers.contains(containerId)) {
+        groupId = request.groupId;
+      }
+    }
+    return groupId;
+  }
+
+  /**
+   * Returns deploy groupId for operator which is to undergo deploy. Operators
+   * undergoing deploy for first time will have groupId as 0
+   * @param operatorId
+   * @return groupId <br/>
+   *         <b>Note:</b> groupId 0 indicates and indipendent event, with no
+   *         group
+   */
+  public long getGroupIdForOperatorToDeploy(int operatorId)
+  {
+    for (DeployRequest requests : getDeployRequests().values()) {
+      if (requests.operatorsToDeploy.contains(operatorId)) {
+        return requests.groupId;
+      }
+    }
+    return 0;
+  }
+
+  /**
+   * Adds operator to deploy. The operator is added to request associated with containerId
+   * @param containerIs
+   * @param operator
+   */
+  public void addOperatorToDeploy(String containerId, PTOperator oper)
+  {
+    DeployRequest request = getDeployRequest(containerId);
+    if (request != null) {
+      request.operatorsToDeploy.add(oper.getId());
+    }
+  }
+
+  /**
+   * Removes operator from deploy request
+   * @param operatorId
+   */
+  public void removeOperatorFromDeployRequest(int operatorId)
+  {
+    for (DeployRequest request : getDeployRequests().values()) {
+      if (request.operatorsToDeploy.contains((operatorId))) {
+        request.operatorsToDeploy.remove(operatorId);
+      }
+    }
+  }
+
+  /**
+   * Create DeployRequest to group deploy/undeploy of related container/operator
+   * events under one groupId to find related events.
+   * To start will all related operators are added to opertorsToUndeploy list,
+   * they will eventually move to operatorsToDeploy when operator undergo redeploy cycle.
+   * @param containerId
+   * @param affectedOperators
+   */
+  public void addNewDeployRequest(String containerId, Set<PTOperator> affectedOperators)
```

Comment:
Ideally, when we call scheduleContainerRestart for a container, the container itself is going to get restarted, so I don't see possibility of having multiple request for same container.